osu.Game/Screens/OnlinePlay/Components/OnlinePlayBackgroundScreen.cs

```diff
@@ -0,0 +1,101 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Threading;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Screens;
+using osu.Game.Online.Rooms;
+using osuTK;
+
+#nullable enable
+
+namespace osu.Game.Screens.OnlinePlay.Components
+{
+    public abstract class OnlinePlayBackgroundScreen : BackgroundScreen
+    {
+        private CancellationTokenSource? cancellationSource;
+        private PlaylistItemBackground? background;
+
+        protected OnlinePlayBackgroundScreen()
+            : base(false)
+        {
+        }
+
+        [BackgroundDependencyLoader]
+        private void load()
+        {
+            switchBackground(new PlaylistItemBackground(playlistItem));
+        }
+
+        private PlaylistItem? playlistItem;
+
+        protected PlaylistItem? PlaylistItem
+        {
+            get => playlistItem;
+            set
+            {
+                if (playlistItem == value)
+                    return;
+
+                playlistItem = value;
+
+                if (LoadState > LoadState.Ready)
+                    updateBackground();
+            }
+        }
+
+        private void updateBackground()
+        {
+            Schedule(() =>
+            {
+                var beatmap = playlistItem?.Beatmap.Value;
+
+                if (background?.BeatmapInfo?.BeatmapSet?.OnlineInfo?.Covers?.Cover == beatmap?.BeatmapSet?.OnlineInfo?.Covers?.Cover)
+                    return;
+
+                cancellationSource?.Cancel();
+                LoadComponentAsync(new PlaylistItemBackground(playlistItem), switchBackground, (cancellationSource = new CancellationTokenSource()).Token);
+            });
+        }
+
+        private void switchBackground(PlaylistItemBackground newBackground)
+        {
+            float newDepth = 0;
+
+            if (background != null)
+            {
+                newDepth = background.Depth + 1;
+                background.FinishTransforms();
+                background.FadeOut(250);
+                background.Expire();
+            }
+
+            newBackground.Depth = newDepth;
+            newBackground.BlurTo(new Vector2(10));
+
+            AddInternal(background = newBackground);
+        }
+
+        protected override void Update()
+        {
+            base.Update();
+
+            // This is a static screen, so override the scale set in base.Update(), but also the scale set by the screen stack.
+            Scale = new Vector2(1f / BackgroundScreenStack.BACKGROUND_SCALE);
```

Comment:
Where can I see this taking effect? Even when reading the inline comment and commenting this out I can't seem to notice the issue (or figure what this is trying to achieve).

Reply:
Comment this out and open `TestSceneMultiplayer` at 10% speed, you'll see the background poking out from the top into the waves.

Reply:
Do you mean this?

![20210823 134743 (dotnet)](https://user-images.githubusercontent.com/191335/130391352-72b82929-6885-473b-9da6-03860fa1bb45.png)

It seems to happen even with the scale applied, just a bit less.

Reply:
This is how it looks on your branch currently in game testing:

https://user-images.githubusercontent.com/191335/130391484-623bb803-f177-4fd5-bbda-de55226912e1.mp4

It gets less noticeable with taller windows, it would seem.

Reply:
That looks like you've just set the scale to `Vector2.One`. There are two places where scale is applied - one in `BackgroundScreenStack` with value `1.06f` which is what I assume you're seeing there, and a second time in this class' `base.Update()` which is much larger than that.

I don't know where you're getting your results from, this is how it looks for me:
![image](https://user-images.githubusercontent.com/1329837/130391760-cd19c44a-6dfe-4889-80d5-bed01694b956.png)
![image](https://user-images.githubusercontent.com/1329837/130391785-a3b102fa-eeae-4217-a9e7-378f7d005701.png)

Reply:
I'm just testing on your branch with no changes, running the game using the `osu.Desktop` project.

Here's two window sizes:

## Tall

![20210823 135618 (dotnet)](https://user-images.githubusercontent.com/191335/130392456-73f82d97-1fd0-4f89-a27b-7f8b7ce931f5.png)

## Wide (as per video)

![20210823 135651 (dotnet)](https://user-images.githubusercontent.com/191335/130392502-3ccaf308-d230-4e3e-baaa-4c86e02948fb.png)

Reply:
I can't repro this. I've tested both visual tests and in-game at both resolutions. It's either something on your end, or something you need to investigate.

Reply:
As discussed, this is because you didn't have parallax enabled. I've moved the gradient to the background itself to fix this issue in a more global way. Note that with parallax enabled, this results in the background screen extending further above the actual lounge screens than expected. In an optimal world we would be able to turn on masking to avoid this, but due to the aforementioned bug this is not done yet.

Probably something to reconsider if it ever becomes an issue in the future. Right now it's almost indiscernible (if you look closely at the wave container effect you'll notice that it's not as visible as usual, but it all happens very fast).